manifests/config.pp

```diff
@@ -3,114 +3,85 @@
-      'api_addr'          => $::vault::api_addr,
+        'listener'          => $vault::listener,
+        'storage'           => $vault::storage,
+        'ha_storage'        => $vault::ha_storage,
+        'seal'              => $vault::seal,
+        'telemetry'         => $vault::telemetry,
+        'disable_cache'     => $vault::disable_cache,
+        'default_lease_ttl' => $vault::default_lease_ttl,
+        'max_lease_ttl'     => $vault::max_lease_ttl,
+        'disable_mlock'     => $vault::disable_mlock,
+        'ui'                => $vault::enable_ui,
+        'api_addr'          => $vault::api_addr,
     })
 
-    $config_hash = merge($_config_hash, $::vault::extra_config)
+    $config_hash = merge($_config_hash, $vault::extra_config)
 
-    file { "${::vault::config_dir}/config.json":
+    file { "${vault::config_dir}/config.json":
       content => to_json_pretty($config_hash),
-      owner   => $::vault::user,
-      group   => $::vault::group,
-      mode    => $::vault::config_mode,
+      owner   => $vault::user,
+      group   => $vault::group,
+      mode    => $vault::config_mode,
     }
 
     # If manage_storage_dir is true and a file or raft storage backend is
     # configured, we create the directory configured in that backend.
     #
-    if $::vault::manage_storage_dir {
-
-      if $::vault::storage['file'] {
+    if $vault::manage_storage_dir {
+      if $vault::storage['file'] {
         $_storage_backend = 'file'
-      } elsif $::vault::storage['raft'] {
+      } elsif $vault::storage['raft'] {
         $_storage_backend = 'raft'
       } else {
         fail('Must provide a valid storage backend: file or raft')
       }
 
-      if $::vault::storage[$_storage_backend]['path'] {
-        file { $::vault::storage[$_storage_backend]['path']:
+      if $vault::storage[$_storage_backend]['path'] {
+        file { $vault::storage[$_storage_backend]['path']:
           ensure => directory,
-          owner  => $::vault::user,
-          group  => $::vault::group,
+          owner  => $vault::user,
+          group  => $vault::group,
         }
       } else {
         fail("Must provide a path attribute to storage ${_storage_backend}")
       }
-
     }
   }
 
   # If nothing is specified for manage_service_file, defaults will be used
   # depending on the install_method.
   # If a value is passed, it will be interpretted as a boolean.
-  if $::vault::manage_service_file == undef {
-    case $::vault::install_method {
-      'archive': { $real_manage_service_file = true  }
+  if $vault::manage_service_file == undef {
+    case $vault::install_method {
+      'archive': { $real_manage_service_file = true }
       'repo':    { $real_manage_service_file = false }
       default:   { $real_manage_service_file = false }
     }
   } else {
-    validate_bool($::vault::manage_service_file)
-    $real_manage_service_file = $::vault::manage_service_file
+    validate_bool($vault::manage_service_file)
```

Comment:
validate_* is also deprecated.

Reply:
removed it

Reply:
https://github.com/voxpupuli/puppet-vault/pull/1/commits/935075152d10d3136fee68326af05f8bc183e400